File: baseproject-android/app/src/main/java/com/chang/template/net/BaseResponse.java
package com.chang.template.net;

/**
 * Created by Howard Chang on 2017/2/15
 */
public class BaseResponse<T> {

    private boolean show_message;
    private Integer error;
    private String message;
    protected T data;

    public void setData(T data) {
        this.data = data;
    }

    public int getError() {
        return error != null ? error : -1;
    }

    public String getMessage() {
        return message != null ? message : "";
    }

    public boolean isShowMessage() {
        return show_message;
    }

    public T getData() {
        return data;
    }
}

File: baseproject-android/app/src/main/java/com/chang/template/net/BaseResponse.kt
package com.chang.template.net

/**
 * Created by Howard Chang on 2017/2/15
 */
open class BaseResponse<T> {

    val isShowMessage: Boolean = false
    private val error: Int? = null
    private val message: String? = null

    fun getError(): Int {
        return error ?: -1
    }

    fun getMessage(): String {
        return message ?: ""
    }
}
